.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,23 @@
+Thank you for contributing to JanusGraph.
+
+In order to streamline the review of the contribution we ask you
+to ensure the following steps have been taken:
+
+### For all changes:
+- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
+- [ ] Does your PR body contains #xyz where xyz is the issue number you are trying to resolve?
+- [ ] Has your PR been rebased against the latest commit within the target branch (typically ```master```)?
+- [ ] Is your initial contribution a single, squashed commit?
```

Comment:
Is this saying it's necessary to have a single squashed commit for all PRs? I think it is sometimes helpful to separate out commits. Examples include multiple authors collaborating on a PR, separating documentation and formatting updates from feature updates and more generally for larger features where the work has been broken up into meaningful commits.